homeassistant/components/mqtt/config_flow.py

```diff
@@ -167,6 +167,24 @@
 PWD_NOT_CHANGED = "__**password_not_changed**__"
 
 
+@callback
+def update_password_from_user_input(
+    entry_password: str | None, user_input: dict[str, Any]
+) -> dict[str, Any]:
+    """Update the password if the entry has been updated."""
```

Comment:
Maybe explain in the doc string that the goal it to avoid reflecting the stored password back to the UI?